test/boost/cdc_test.cc

```diff
@@ -643,19 +646,29 @@ static void test_collection(cql_test_env& e, data_type val_type, std::vector<col
         if (t.prev.is_null()) {
             BOOST_REQUIRE(pre_image.empty());
         } else {
-            BOOST_REQUIRE_GT(pre_image.size(), 0);
-            auto val = *pre_image.back()[val_index];
-            BOOST_REQUIRE_EQUAL(t.prev, f(value_cast<tuple_type_impl::native_type>(col_type->deserialize(bytes_view(val))).at(1)));
+            BOOST_REQUIRE_GE(pre_image.size(), t.changes.size());
         }
 
-        auto val = *updates.back()[val_index];
-        auto tup = value_cast<tuple_type_impl::native_type>(col_type->deserialize(bytes_view(val)));
+        BOOST_REQUIRE_GE(updates.size(), t.changes.size());
 
-        if (!t.next.is_null()) {
-            BOOST_REQUIRE_EQUAL(t.next, f(tup.at(1)));
-        }
+        for (size_t change_ix = 0; change_ix < t.changes.size(); ++change_ix) {
+            auto& change = t.changes[change_ix];
```

Comment:
Iterators. Indexes are so C89

Reply:
I need the `change_ix` in multiple places below.
Could use an iterator and then subtract `changes.begin()` to get `change_ix`. Is that what you'd prefer?